wodles/aws/wazuh_integration.py

```diff
@@ -0,0 +1,330 @@
+from datetime import datetime
+from datetime import timezone
+
+import aws_tools
+
+sys.path.insert(0, path.dirname(path.dirname(path.abspath(__file__))))
+import utils
+
+DEPRECATED_TABLES = {'log_progress', 'trail_progress'}
+DEFAULT_GOV_REGIONS = {'us-gov-east-1', 'us-gov-west-1'}
+SERVICES_REQUIRING_REGION = {'inspector', 'cloudwatchlogs'}
+WAZUH_DEFAULT_RETRY_CONFIGURATION = {'max_attempts': 10, 'mode': 'standard'}
+MESSAGE_HEADER = "1:Wazuh-AWS:"
+
+
+class WazuhIntegration:
+    """
+    Class with common methods
+    :param access_key: AWS access key id
+    :param secret_key: AWS secret access key
+    :param profile: AWS profile
+    :param iam_role_arn: IAM Role
+    :param service_name: Name of the service (s3 for services which stores logs in buckets)
+    :param region: Region of service
+    :param iam_role_duration: The desired duration of the session that is going to be assumed.
+    :param external_id: AWS external ID for IAM Role assumption
+    """
+
+    def __init__(self, access_key, secret_key, profile, iam_role_arn, service_name=None, region=None,
+                 discard_field=None, discard_regex=None, sts_endpoint=None,
+                 service_endpoint=None, iam_role_duration=None, external_id=None):
+
+        self.wazuh_path = utils.find_wazuh_path()
+        self.wazuh_version = utils.get_wazuh_version()
+        self.wazuh_queue = f'{self.wazuh_path}/queue/sockets/queue'
+        self.wazuh_wodle = f'{self.wazuh_path}/wodles/aws'
+
+        self.connection_config = self.default_config()
+        self.client = self.get_client(access_key=access_key,
+                                      secret_key=secret_key,
+                                      profile=profile,
+                                      iam_role_arn=iam_role_arn,
+                                      service_name=service_name,
+                                      region=region,
+                                      sts_endpoint=sts_endpoint,
+                                      service_endpoint=service_endpoint,
+                                      iam_role_duration=iam_role_duration,
+                                      external_id=external_id
+                                      )
+
+        self.discard_field = discard_field
+        self.discard_regex = re.compile(fr'{discard_regex}')
+        # to fetch logs using this date if no only_logs_after value was provided on the first execution
+        self.default_date = datetime.utcnow().replace(hour=0, minute=0, second=0, microsecond=0, tzinfo=timezone.utc)
+
+    @staticmethod
+    def default_config():
+        args = {}
+        if not path.exists(aws_tools.DEFAULT_AWS_CONFIG_PATH):
+            args['config'] = botocore.config.Config(retries=WAZUH_DEFAULT_RETRY_CONFIGURATION.copy())
+            aws_tools.debug(
+                f"Generating default configuration for retries: mode {args['config'].retries['mode']} - max_attempts {args['config'].retries['max_attempts']}",
```

Comment:
Too long line, fix this everywhere you find it (when using a shorter line it's possible).